components/ntp_sponsored_images/ntp_sponsored_images_component_manager.cc

```diff
@@ -0,0 +1,224 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ntp_sponsored_images/ntp_sponsored_images_component_manager.h"
+
+#include <algorithm>
+#include <vector>
+
+#include "base/bind.h"
+#include "base/files/file_util.h"
+#include "base/json/json_reader.h"
+#include "base/strings/string_util.h"
+#include "base/strings/string_split.h"
+#include "base/task/post_task.h"
+#include "base/values.h"
+#include "brave/components/ntp_sponsored_images/locale_helper.h"
+#include "brave/components/ntp_sponsored_images/ntp_sponsored_images_data.h"
+#include "brave/components/ntp_sponsored_images/ntp_sponsored_images_internal_data.h"
+#include "brave/components/ntp_sponsored_images/regional_component_data.h"
+#include "brave/components/ntp_sponsored_images/switches.h"
+#include "content/public/browser/browser_context.h"
+
+namespace {
+
+constexpr char kDefaultRegion[] = "US";
+
+constexpr char kPhotoJsonFilename[] = "photo.json";
+constexpr char kComponentName[] = "NTP sponsored images";
+
+constexpr char kLogoImageURLPath[] = "logo.imageUrl";
+constexpr char kLogoAltPath[] = "logo.alt";
+constexpr char kLogoCompanyNamePath[] = "logo.companyName";
+constexpr char kLogoDestinationURLPath[] = "logo.destinationUrl";
+constexpr char kWallpapersPath[] = "wallpapers";
+constexpr char kWallpaperImageURLPath[] = "imageUrl";
+
+const std::string GetRegionCode(const std::string& locale) {
+  std::vector<std::string> locale_components = base::SplitString(locale, ".",
+      base::KEEP_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+
+  if (locale_components.size() == 0) {
+    return kDefaultRegion;
+  }
+
+  auto normalized_locale = locale_components.front();
+  std::replace(normalized_locale.begin(), normalized_locale.end(), '-', '_');
+
+  std::vector<std::string> components = base::SplitString(
+      normalized_locale, "_", base::KEEP_WHITESPACE, base::SPLIT_WANT_ALL);
+
+  if (components.size() != 2) {
+    return kDefaultRegion;
+  }
+
+  auto region_code = components.at(1);
+  return region_code;
+}
+
+std::string ReadPhotoJsonData(const base::FilePath& photo_json_file_path) {
+  std::string contents;
+  bool success = base::ReadFileToString(photo_json_file_path, &contents);
+  if (!success || contents.empty()) {
+    DVLOG(2) << "ReadPhotoJsonData: cannot "
+             << "read photo.json file " << photo_json_file_path;
+  }
+  return contents;
+}
+
+}  // namespace
+
+NTPSponsoredImagesComponentManager::NTPSponsoredImagesComponentManager(
+    BraveComponent::Delegate* delegate,
+    component_updater::ComponentUpdateService* cus)
+    : BraveComponent(delegate),
+      weak_factory_(this) {
+  // Early return for test.
+  if (!cus)
+    return;
+
+#if defined(OFFICIAL_BUILD)
```

Comment:
is this backwards? should it be `if !defined(OFFICIAL_BUILD)`?

Reply:
er yeah but also I thought I deleted this since we got the go-ahead to do this on all builds 🤷‍♂

Reply:
Removed now! Thanks for spotting